### PR TITLE
cli: snapshot stale working copy before updating

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: tests/test_workspaces.rs [hunk 1]
```diff
@@ -122,17 +122,21 @@ fn test_workspaces_conflicting_edits() {
     Hint: Run `jj workspace update-stale` to update it.
     "###);
     let stdout = test_env.jj_cmd_success(&secondary_path, &["workspace", "update-stale"]);
-    // It was detected that the working copy is now stale
-    // TODO: Since there was an uncommitted change in the working copy, it should
+    // It was detected that the working copy is now stale.
+    // Since there was an uncommitted change in the working copy, it should
     // have been committed first (causing divergence)
     insta::assert_snapshot!(stdout, @r###"
+    Concurrent modification detected, resolving automatically.
```
Comment (martinvonz) [inline tests/test_workspaces.rs:129]: Do you think it would be useful to also show the operation log?

**Review by martinvonz — APPROVED** [review]
Thanks for fixing!

**Comment by martinvonz** [comment]
> If I understand correctly, this makes the snapshot commit look the same as the old and probably good revision. Perhaps jj should make it clear which is which and encourage the user to abandon the snapshot commit after checking that the probably good revision is, in fact, good. In my experience, that's the right thing to do 90% of the time.

We probably want a general solution for resolving divergence, but it seems that it would still be useful to print something like "Working copy saved as commit abc123".

> One way to do this would be to change the description of the recovered workplace state. E.g. prepend `(Recovered stale workspace state) ` to it.

That also seems useful.

File: tests/test_workspaces.rs (hunk 1 above)
Comment (yuja) [reply tests/test_workspaces.rs:129]: Might be useful, but I think showing a hint like "(Run `jj op log`)" is good enough.

**Comment by yuja** [comment]
> > We probably want a general solution for resolving divergence, but it seems that it would still be useful to print something like "Working copy saved as commit abc123".
> 
> Only for snapshot of stale working copy? or always?
> 
> I think the latter would be too verbose.

Anyway, I'll address this as a follow up. Since the stale wc commit is rebased by op heads resolution, we can't print the commit id on shapshot.

**Pull request merged**